flash/tabular/regression/data.py

```diff
@@ -403,3 +404,114 @@ def from_dicts(
+        predict_list: List[Union[tuple, dict]] = None,
+        input_cls: Type[Input] = TabularRegressionListInput,
+        transform: INPUT_TRANSFORM_TYPE = InputTransform,
+        transform_kwargs: Optional[Dict] = None,
+        **data_module_kwargs: Any,
+    ) -> "TabularRegressionData":
+        """Creates a :class:`~flash.tabular.regression.data.TabularRegressionData` object from the given data (in
+        the form of list of a tuple or a dictionary).
+
+        .. note::
+
+            The ``categorical_fields``, ``numerical_fields``, and ``target_field`` do not need to be provided if
+            ``parameters`` are passed instead. These can be obtained from the
+            :attr:`~flash.tabular.data.TabularData.parameters` attribute of the
+            :class:`~flash.tabular.data.TabularData` object that contains your training data.
+
+        The targets will be extracted from the ``target_field`` in the data frames.
+        To learn how to customize the transforms applied for each stage, read our
+        :ref:`customizing transforms guide <customizing_transforms>`.
+
+        Args:
+            categorical_fields: The fields (column names) in the dictionary containing categorical data.
+            numerical_fields: The fields (column names) in the dictionary containing numerical data.
+            target_field: The field (column name) in the dictionary containing the targets.
+            parameters: Parameters to use if ``categorical_fields``, ``numerical_fields``, and ``target_field`` are not
+                provided (e.g. when loading data for inference or validation).
+            train_list: The data to use when training.
+            val_list: The data to use when validating.
+            test_list: The data to use when testing.
+            predict_list: The data to use when predicting.
+            input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the data.
+            transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use.
+            transform_kwargs: Dict of keyword arguments to be provided when instantiating the transforms.
+            data_module_kwargs: Additional keyword arguments to provide to the
+              :class:`~flash.core.data.data_module.DataModule` constructor.
+
+        Returns:
+            The constructed :class:`~flash.tabular.regression.data.TabularRegressionData`.
+
+        Examples
+        ________
+
+        .. testsetup::
+
+            >>> train_data = [
+            ...     {"animal": "cat", "friendly": "yes", "weight": 6},
+            ...     {"animal": "dog", "friendly": "yes", "weight": 10},
+            ...     {"animal": "cat", "friendly": "no", "weight": 5},
+            ... ]
+            >>> predict_data = [
+            ...     {"friendly": "yes", "weight": 7},
+            ...     {"friendly": "no", "weight": 12},
+            ...     {"friendly": "yes", "weight": 5},
+            ... ]
```

Comment:
This block won't be visible in the rendered docs. You could either change it to just a `doctest` block or jsut provide the data literals when creating the datamodule below. Same applies to the classification docstring 😃

Reply:
Ah, good observation! I just pushed commit to fixing it for `from_dicts` and `from_lists` (previous PR as well). I'll check with the rendered doc once CI is green.

Reply:
This seems to be fixed now. Please take a look at the PR description for relevant links to the docs :)